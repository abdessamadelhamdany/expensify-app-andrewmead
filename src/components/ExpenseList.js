import React from "react";
import { connect } from "react-redux";

const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    {props.expenses.map((expense) => (
      <div key={expense.id}>
        <h3>{expense.description}</h3>
      </div>
    ))}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
  };
};

export default connect(mapStateToProps)(ExpenseList);
