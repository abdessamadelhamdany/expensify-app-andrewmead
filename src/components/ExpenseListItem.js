import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { removeExpense } from "../actions/expenses";
// console.log(now.format("MMM Do, YYYY"));

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <h3>{description}</h3>
    <p>
      {amount} - {createdAt}
    </p>
    <Link to={`/edit/${id}`}>Edit</Link>
    <button
      onClick={() => {
        dispatch(removeExpense({ id }));
      }}
    >
      Remove
    </button>
  </div>
);

export default connect()(ExpenseListItem);
