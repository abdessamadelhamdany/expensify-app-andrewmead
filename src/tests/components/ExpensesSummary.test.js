import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should correctly render ExpensesSummary with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={9434} />);
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpensesSummary with multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={7} expensesTotal={9425734} />);
  expect(wrapper).toMatchSnapshot();
});
