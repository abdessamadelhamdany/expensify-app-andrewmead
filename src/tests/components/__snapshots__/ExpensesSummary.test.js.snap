// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expenses
     totalling 
    $94.34
  </h2>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    7
     
    expense
     totalling 
    $94,257.34
  </h2>
</div>
`;
